File: CS/DockingDemo.Wpf/Helpers/ImagesHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Media;
using System.Windows.Media.Imaging;

namespace DockingDemo.Helpers {
    public static class ImagesHelper {

        public static IEnumerable<ImageSource> NaturePhotos { get; private set; }
        public static ImageSource NaturePhoto1 { get { return NaturePhotos.ElementAt(0); } }
        public static ImageSource NaturePhoto2 { get { return NaturePhotos.ElementAt(1); } }
        public static ImageSource NaturePhoto3 { get { return NaturePhotos.ElementAt(2); } }
        public static ImageSource NaturePhoto4 { get { return NaturePhotos.ElementAt(3); } }
        public static ImageSource NaturePhoto5 { get { return NaturePhotos.ElementAt(4); } }
        public static ImageSource NaturePhoto6 { get { return NaturePhotos.ElementAt(5); } }
        public static ImageSource NaturePhoto7 { get { return NaturePhotos.ElementAt(6); } }
        public static ImageSource NaturePhoto8 { get { return NaturePhotos.ElementAt(7); } }
        public static ImageSource NaturePhoto9 { get { return NaturePhotos.ElementAt(8); } }
        public static ImageSource NaturePhoto10 { get { return NaturePhotos.ElementAt(9); } }
        public static ImageSource GetRandomNaturePhoto() {
            return NaturePhotos.ElementAt(rnd.Next(NaturePhotos.Count() - 1));
        }

        static ImagesHelper() {
            string natureUriPath = @"/DockingDemo;component/Images/Photos/Nature/";
            Func<string, ImageSource> getNatureImage = x => new BitmapImage(new Uri(natureUriPath + x, UriKind.Relative));
            var naturePhotos = new List<ImageSource>();
            naturePhotos.Add(getNatureImage("01.JPG"));
            naturePhotos.Add(getNatureImage("02.JPG"));
            naturePhotos.Add(getNatureImage("03.JPG"));
            naturePhotos.Add(getNatureImage("04.JPG"));
            naturePhotos.Add(getNatureImage("05.JPG"));
            naturePhotos.Add(getNatureImage("06.JPG"));
            naturePhotos.Add(getNatureImage("07.JPG"));
            naturePhotos.Add(getNatureImage("08.JPG"));
            naturePhotos.Add(getNatureImage("09.JPG"));
            naturePhotos.Add(getNatureImage("10.JPG"));
            NaturePhotos = naturePhotos;
        }
        static readonly Random rnd = new Random();
    }
}

File: VB/DockingDemo.Wpf/Helpers/ImagesHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Media
Imports System.Windows.Media.Imaging

Namespace DockingDemo.Helpers
    Public NotInheritable Class ImagesHelper

        Private Sub New()
        End Sub


        Private Shared privateNaturePhotos As IEnumerable(Of ImageSource)
        Public Shared Property NaturePhotos() As IEnumerable(Of ImageSource)
            Get
                Return privateNaturePhotos
            End Get
            Private Set(ByVal value As IEnumerable(Of ImageSource))
                privateNaturePhotos = value
            End Set
        End Property
        Public Shared ReadOnly Property NaturePhoto1() As ImageSource
            Get
                Return NaturePhotos.ElementAt(0)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto2() As ImageSource
            Get
                Return NaturePhotos.ElementAt(1)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto3() As ImageSource
            Get
                Return NaturePhotos.ElementAt(2)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto4() As ImageSource
            Get
                Return NaturePhotos.ElementAt(3)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto5() As ImageSource
            Get
                Return NaturePhotos.ElementAt(4)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto6() As ImageSource
            Get
                Return NaturePhotos.ElementAt(5)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto7() As ImageSource
            Get
                Return NaturePhotos.ElementAt(6)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto8() As ImageSource
            Get
                Return NaturePhotos.ElementAt(7)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto9() As ImageSource
            Get
                Return NaturePhotos.ElementAt(8)
            End Get
        End Property
        Public Shared ReadOnly Property NaturePhoto10() As ImageSource
            Get
                Return NaturePhotos.ElementAt(9)
            End Get
        End Property
        Public Shared Function GetRandomNaturePhoto() As ImageSource
            Return NaturePhotos.ElementAt(rnd.Next(NaturePhotos.Count() - 1))
        End Function

        Shared Sub New()
            Dim natureUriPath As String = "/DockingDemo;component/Images/Photos/Nature/"
            Dim getNatureImage As Func(Of String, ImageSource) = Function(x) New BitmapImage(New Uri(natureUriPath & x, UriKind.Relative))

            Dim naturePhotos_Renamed = New List(Of ImageSource)()
            naturePhotos_Renamed.Add(getNatureImage("01.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("02.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("03.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("04.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("05.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("06.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("07.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("08.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("09.JPG"))
            naturePhotos_Renamed.Add(getNatureImage("10.JPG"))
            NaturePhotos = naturePhotos_Renamed
        End Sub
        Private Shared ReadOnly rnd As New Random()
    End Class
End Namespace
